src/components/Layout/ChipBase.js

```diff
@@ -1,48 +1,94 @@
 import { DimensionIcon } from '../MainSidebar/DimensionItem/DimensionIcon.js'
 import styles from './styles/Chip.module.css'
 
+const VALUE_TYPE_TRUE_ONLY_NUM_OPTIONS = 2
+const VALUE_TYPE_BOOLEAN_NUM_OPTIONS = 3
+
 // Presentational component used by dnd - do not add redux or dnd functionality
 
-const renderChipLabelSuffix = (items, numberOfConditions) => {
-    let itemsLabel = ''
-    if (items.length) {
-        itemsLabel = i18n.t('{{count}} selected', {
-            count: items.length,
+const getChipSuffix = ({
+    dimension,
+    conditions = {},
+    items = [],
+    metadata,
+}) => {
+    const { id, dimensionType, optionSet, valueType } = dimension
+
+    if (
+        (id === DIMENSION_ID_ORGUNIT ||
+            dimensionType === DIMENSION_TYPE_PERIOD) &&
+        !items.length
+    ) {
+        return ''
+    }
+
+    const all = i18n.t('all')
+
+    if (!conditions.condition && !conditions.legendSet && !items.length) {
+        return `: ${all}`
+    }
+
+    const conds = getConditions({ conditions, metadata, dimension })
+
+    if (
+        (valueType === VALUE_TYPE_TRUE_ONLY &&
+            conds.length === VALUE_TYPE_TRUE_ONLY_NUM_OPTIONS) ||
+        (valueType === VALUE_TYPE_BOOLEAN &&
+            conds.length === VALUE_TYPE_BOOLEAN_NUM_OPTIONS)
+    ) {
+        return `: ${all}`
+    }
+
+    const numberOfConditions = conds.length
+
+    let suffix = ''
+    if (optionSet || items.length) {
+        const selected = items.length || numberOfConditions
+
+        suffix = i18n.t('{{count}} selected', {
+            count: selected,
+            defaultValue: '{{count}} selected',
+            defaultValue_plural: '{{count}} selected',
         })
     } else if (numberOfConditions) {
-        itemsLabel = i18n.t('{{count}} conditions', {
+        suffix = i18n.t('{{count}} conditions', {
             count: numberOfConditions,
             defaultValue: '{{count}} condition',
             defaultValue_plural: '{{count}} conditions',
         })
     }
-    return itemsLabel ? `: ${itemsLabel}` : ''
+    return suffix ? `: ${suffix}` : ''
 }
 
-export const ChipBase = ({
-    dimensionName,
-    dimensionType,
-    items,
-    numberOfConditions,
-}) => (
-    <div className={cx(styles.chipBase)}>
-        <div className={styles.leftIcon}>
-            <DimensionIcon dimensionType={dimensionType} />
+export const ChipBase = (props) => {
+    const { name, dimensionType, stageName } = props.dimension
+    return (
+        <div className={cx(styles.chipBase)}>
+            <div className={styles.leftIcon}>
+                <DimensionIcon dimensionType={dimensionType} />
+            </div>
+            <span className={styles.label}>
+                <span className={styles.primary}>{name}</span>
+                {stageName && (
+                    <span className={styles.secondary}>{stageName}</span>
+                )}
+            </span>
+            <span className={styles.suffix}>{getChipSuffix(props)}</span>
```

Comment:
Unless there's a specific reason for keeping the `getChipSuffix` function outside of the `ChipBase` scope, we might as well just include inside `ChipBase` and get the props directly from `ChipBase` instead of passing them in (then you can do the regular props destruction in the input params as we usually do).

`export const ChipBase = ({foo, bar, baz}) => { const getChipSuffix = () => ...`

Reply:
ok